Automattic/jetpack
# Conversation/Dialogue: use `parent` block settings to limit Dialogue availability

This PR removes the custom template-lock system since we consider it could bring performance issues. Using the `parent` block setting property will be ok to provide a lock system for the Dialogue block for v1.
We wanted to make the Dialogue block available also for deeper nesting levels but finally decided to remove it, at least for this first implementation.

Fixes #

#### Changes proposed in this Pull Request:

* Conversation/Conversation: use `parent` block settings to limit Dialogue availability

#### Jetpack product discussion


p1613137937245000-slack-

#### Does this pull request change what data or activity we track or use?



#### Testing instructions:






* Create a new post
* Confirm the `Dialogue` is not available from the blocks picker
<img src="https://user-images.githubusercontent.com/77539/107781884-2cefd780-6d27-11eb-85c5-84a056fdc5d0.png" width="600px" />

* Create a `Conversation` block
* Confirm the `Dialogue` block is properly rendered
* Confirm you can add/remove/edit the `Dialogue` blocks
* Confirm you can add other blocks into the `Conversation` block.

![image](https://user-images.githubusercontent.com/77539/107782063-6294c080-6d27-11eb-8022-6ef051d9a473.png)


#### Proposed changelog entry for your changes:


*
